### PR TITLE
Print non-breaking space directly in lint.yml

### PR DESCRIPTION
**Pull request by driazati**
After some fun investigating, @samestep found that `\u1234` to produce a unicode character is only supported in bash > 4.2, but MacOS ship with bash/sh 3.2, so it was searching for the literal string `u1234`. This fixes the issue by printing out the char directly via its UTF-8 bytes and `printf`.

Differential Revision: [D27952866](https://our.internmc.facebook.com/intern/diff/27952866/)

### CODE REVIEW

File: tools/actions_local_runner.py [hunk 1]
```diff
@@ -60,7 +60,8 @@ def find_changed_files(merge_base: str = "origin/master") -> List[str]:
     cached = git(["diff", "--cached", "--name-only"])
 
     # Committed
-    diff_with_origin = git(["diff", "--name-only", "--merge-base", merge_base, "HEAD"])
+    merge_base = git(["merge-base", "origin/master", "HEAD"])[0]
```
Comment (samestep) [inline tools/actions_local_runner.py:63]: Given this line, what is the purpose of the `merge_base` parameter to this function now?

Comment (driazati) [reply tools/actions_local_runner.py:63]: None really, I took it out in the latest push

**Review by samestep — APPROVED** [review]

**Comment by samestep** [comment]
> This was detecting its own non-breaking space character with BSD grep on Macs

Could you explain what exactly this means? To clarify, the `.github/workflows/lint.yml` doesn't _actually_ contain a non-breaking space character, does it?

**Comment by driazati** [comment]
> > This was detecting its own non-breaking space character with BSD grep on Macs
> 
> Could you explain what exactly this means? To clarify, the `.github/workflows/lint.yml` doesn't _actually_ contain a non-breaking space character, does it?

Right, it doesn't. I still don't really know what is actually going on, but from checking it out it looks like it ends up firing for the string `\u00a0` anywhere in a file. The confusing bit is that even GNU grep (via `brew install grep`) still finds that line on MacOS but not Linux:

```
$ ggrep -r -In $'\u00a0' .github
.github/workflows/lint.yml:66:          (! git --no-pager grep -In $'\u00a0' -- . || (echo "The above lines have non-breaking spaces (U+00A0); please convert them to spaces (U+0020)"; false))
```

**Comment by samestep** [comment]
> The confusing bit is that even GNU grep (via brew install grep) still finds that line on MacOS but not Linux:

Oh wow that's crazy haha 😅 good call just disabling it on the file, then

**Comment by samestep** [comment]
For the curious: the underlying problem was not due to differences between different `grep` versions, but was rather due to the fact that macOS comes with Bash 3.2 by default instead of a more recent Bash, and [Bash 3.2 does not support Unicode escape sequences](https://stackoverflow.com/questions/602912/how-do-you-echo-a-4-digit-unicode-character-in-bash#comment19509228_602974):

> Support for \u was added in Bash 4.2.

Thus, we're solving this problem by instead writing out the nbsp character as its UTF-8 bytes via `printf`, which should work on old and new Bash, as well as Zsh etc.

The reason this bug wasn't discovered earlier is that macOS uses Zsh by default instead of Bash, so running the command locally in a terminal seems to work fine; you specifically need to open a `bash` subshell to reproduce the failure. Or, in this case, use [`asyncio.create_subprocess_shell`](https://docs.python.org/3/library/asyncio-subprocess.html#asyncio.create_subprocess_shell).

**Pull request closed without merging**